nifi-nar-bundles/nifi-snmp-bundle/nifi-snmp-processors/src/main/java/org/apache/nifi/snmp/processors/ListenTrapSNMP.java

```diff
@@ -65,14 +74,32 @@ public class ListenTrapSNMP extends AbstractSessionFactoryProcessor {
             .addValidator(StandardValidators.PORT_VALIDATOR)
             .build();
 
-    public static final PropertyDescriptor SNMP_USM_USERS_FILE_PATH = new PropertyDescriptor.Builder()
+    public static final PropertyDescriptor SNMP_USM_USERS_JSON_FILE = new PropertyDescriptor.Builder()
             .name("snmp-usm-users-file-path")
-            .displayName("SNMP Users File Path")
+            .displayName("SNMP Users JSON File")
             .description("The path of the json file containing the user credentials for SNMPv3. Check Usage for more details.")
-            .required(true)
-            .defaultValue("")
+            .required(false)
```

Comment:
I will refactor this part as you suggested creating a new property which determines how we provide the USM data instead of displaying three properties at once. And I'll do a compatibilty check in the end. Thanks.